src/Microsoft.Azure.WebJobs.Extensions.Dapr/Services/DaprServiceListener.cs

```diff
@@ -21,34 +21,49 @@ namespace Microsoft.Azure.WebJobs.Extensions.Dapr.Services
     using Microsoft.Extensions.DependencyInjection;
     using Microsoft.Extensions.Logging;
 
-    sealed class DaprServiceListener : IDisposable, IDaprServiceListener
+    internal sealed class DaprServiceListener : IDisposable, IDaprServiceListener
     {
+        private const string MetadataApiUrl = "";
         readonly HashSet<DaprListenerBase> listeners = new HashSet<DaprListenerBase>();
         readonly HashSet<DaprTopicSubscription> topics = new HashSet<DaprTopicSubscription>(new DaprTopicSubscriptionComparer());
         readonly string appAddress;
+        readonly string daprAddress;
+        readonly bool enableSidecarCheckOnHostStartup;
         readonly ILogger logger;
+        private readonly IDaprClient daprClient;
 
         IWebHost? host;
         int serverStarted;
 
-        public DaprServiceListener(ILoggerFactory loggerFactory, INameResolver resolver)
+        public DaprServiceListener(ILoggerFactory loggerFactory, IDaprClient daprClient, INameResolver resolver)
         {
             this.logger = loggerFactory.CreateLogger(LogCategories.CreateTriggerCategory("Dapr"));
-            this.appAddress = GetDefaultAppAddress(resolver);
+            this.daprClient = daprClient;
+
+            this.appAddress = GetAppAddress(resolver);
+            this.daprAddress = DaprServiceClient.GetDaprHttpAddress(this.logger, resolver);
+            this.enableSidecarCheckOnHostStartup = EnableSidecarCheckOnHostStartup(resolver);
         }
 
         public void Dispose() => this.host?.Dispose();
 
-        static string GetDefaultAppAddress(INameResolver resolver)
+        static string GetAppAddress(INameResolver resolver)
         {
-            if (!int.TryParse(resolver.Resolve("DAPR_APP_PORT"), out int appPort))
+            if (!int.TryParse(resolver.Resolve(Constants.EnvironmentKeys.AppPort), out int appPort))
             {
                 appPort = 3001;
             }
 
             return $"http://127.0.0.1:{appPort}";
         }
 
+        static bool EnableSidecarCheckOnHostStartup(INameResolver resolver)
+        {
+            return bool.TryParse(resolver.Resolve(Constants.EnvironmentKeys.EnableSidecarMetadataCheck), out bool enableSidecarCheck)
+                ? enableSidecarCheck
+                : false; // default
```

Comment:
I think we should keep the FF enabled by default as we would like to always check if configuration is correct or not. 

This is something that should happen until user disables it. My suggestion would be to rename the FF to DISABLE_DAPR_SIDECAR_METADATA_CHECK, and then have default value as false.

Reply:
> we would like to check the configuration of user by default

Is that the case? I don't think we want to enable this check for any other offering besides Functions on CApps. 

In Functions on CApps also, end-users won't be configuring this directly, but the function platform will set this env variable (just like how they are handling `DAPR_APP_PORT`).

Reply:
In Functions on CApp, end-users won't be configuring this directly, but the function platform will set this env variable (just like how they are handling DAPR_APP_PORT)

Reply:
I reversed the flag.